emr/application/controllers/Screeners.php

```diff
@@ -140,58 +140,108 @@ public function added_time() {
       $data['last_name'] = $this->ion_auth->user()->row()->last_name;
       $data['employeeid'] = $this->ion_auth->user()->row()->employeeid;
       $data['email'] = $this->ion_auth->user()->row()->email;
-
+    
       // Getting Dates
       $morn_times = $this->input->post('morn_times');
+      if ($morn_times != null) {
+        $morn_time = $morn_times;
+      } else {
+        $morn_time = [];
+      }
+
       $eve_times = $this->input->post('eve_times');
+      
+      if ($eve_times != null) {
+        $eve_time = $eve_times;
+      } else {
+        $eve_time = [];
+      }
+
       $night_times = $this->input->post('night_times');
+      if ($night_times != null) {
+        $night_time = $night_times;
+      } else {
+        $night_time = [];
+      }
 
-      for ($i=0; $i < sizeof($morn_times); $i++){
-        $my_dt = new DateTime($morn_times[$i]);
+      for ($i=0; $i < sizeof($morn_time); $i++){
+        $my_dt = new DateTime($morn_time[$i]);
         $expires_at = $my_dt->modify(' +8 hour');
         $end_date = $expires_at->format('Y-m-d H:i:s');
         
         $morn = array(
           'employeeid' => $data['employeeid'],
           'first_name' => $data['first_name'],
           'last_name' => $data['last_name'],
-          'start' => $morn_times[$i],
+          'start' => $morn_time[$i],
           'end' => $end_date
         );
         $this->db->insert('availability', $morn);
```

Comment:
you shouldn't do database actions in the controller, the model should be used for that instead

Reply:
Just changed this, now its inserting into the db through the Availability_model

Reply:
Sounds good